src/Qwiq.Core.Rest/QueryFolder.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+
+namespace Microsoft.Qwiq.Client.Rest
+{
+    internal class QueryFolder : Qwiq.QueryFolder
+    {
+        internal QueryFolder(QueryHierarchyItem queryFolder, Func<string, QueryHierarchyItem> folderExpansionFunc)
```

Comment:
Looks like you're doing work in the constructor (calling `folderExpansionFunc`) that could `throw`.

Reply:
It's not called in the constructor, it's called in a lazy func which is invoked later.